### PR TITLE
Allow all azure config options to be overridden in config

### PR DESCRIPTION
**Pull request by mikenorgate**
In some scenarios I want to be able to customize the azure cloud config. I suggest that the rke config should be preferred for every setting

### CODE REVIEW

File: cloud-provider.sh [hunk 1]
```diff
@@ -23,13 +28,26 @@ set_azure_config() {
...
 
-  local az_vm_nic=$(az vm nic list -g ${az_resources_group} --vm-name ${az_vm_name} | jq -r .[0].id | cut -d "/" -f 9)
-  local az_subnet_name=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 11)
-  local az_vnet_name=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 9)
-  local az_vnet_resource_group=$(az vm nic show -g ${az_resources_group} --vm-name ${az_vm_name} --nic ${az_vm_nic}| jq -r .ipConfigurations[0].subnet.id| cut -d"/" -f 5)
+  if [ -z "$az_resources_group" ] ; then
+    az_resources_group="$az_vm_resources_group"
+  fi
+
+  local az_vm_nic=$(az vm nic list -g ${az_vm_resources_group} --vm-name ${az_vm_name} | jq -r .[0].id | cut -d "/" -f 9)
```
Comment (kinarashah) [inline cloud-provider.sh]: Looks like we should use `az_resources_group` here? if it's set from L7 
(Similar for L38, 42 and 46)

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/20632

File: cloud-provider.sh (hunk 1 above)
Comment (mikenorgate) [reply cloud-provider.sh]: Good spot, I've updated it

**Comment by alena1108** [comment]
Merge after Rancher v2.2.5 gets released.

**Pull request merged**